pkg/kopia/command/parse_command_output.go

```diff
@@ -327,6 +331,93 @@ func parseKopiaProgressLine(line string, matchOnlyFinished bool) (stats *Snapsho
 	}
 }
 
+// SnapshotRestoreStats is a container for stats parsed from the output of a
+// `kopia snapshot restore` command.
+type SnapshotRestoreStats struct {
+	FilesProcessed  int64
+	SizeProcessedB  int64
+	FilesTotal      int64
+	SizeTotalB      int64
+	ProgressPercent int64
+}
+
+// SnapshotStatsFromSnapshotRestore parses the output of a `kopia snapshot
+// restore` execution for a log of the stats for that execution.
+func SnapshotStatsFromSnapshotRestore(snapRestoreStderrOutput string) (stats *SnapshotRestoreStats) {
+	if snapRestoreStderrOutput == "" {
+		return nil
+	}
+	logs := regexp.MustCompile("[\r\n]").Split(snapRestoreStderrOutput, -1)
+
+	for _, l := range logs {
+		lineStats := parseKopiaSnapshotRestoreProgressLine(l)
+		if lineStats != nil {
+			stats = lineStats
+		}
+	}
+
+	return stats
+}
+
+func parseKopiaSnapshotRestoreProgressLine(line string) (stats *SnapshotRestoreStats) {
+	match := kopiaSnapshotRestorePattern.FindStringSubmatch(line)
```

Comment:
I think it would be a good idea if we can also share (in comments) how exactly this line  looks like.

Reply:
Done.